apps/frontend/src/components/technique/TechniqueTable.tsx

```diff
@@ -0,0 +1,234 @@
+import AssignmentInd from '@mui/icons-material/AssignmentInd';
+import { Dialog, DialogContent, Typography } from '@mui/material';
+import React, { useState } from 'react';
+import { useQueryParams } from 'use-query-params';
+
+import { useCheckAccess } from 'components/common/Can';
+import SuperMaterialTable, {
+  DefaultQueryParams,
+  UrlQueryParamsType,
+} from 'components/common/SuperMaterialTable';
+import { useTechniquesData } from 'hooks/technique/useTechniquesData';
+import useDataApiWithFeedback from 'utils/useDataApiWithFeedback';
+import { FunctionType } from 'utils/utilTypes';
+
+import AssignedInstrumentsTable from './AssignedInstrumentsTable';
+import AssignInstrumentsToTechniques from './AssignInstrumentsToTechniques';
+import CreateUpdateTechnique from './CreateUpdateTechnique';
+import {
+  InstrumentFragment,
+  TechniqueFragment,
+  UserRole,
+} from '../../generated/sdk';
+
+const columns = [
+  {
+    title: 'Name',
+    field: 'name',
+  },
+  {
+    title: 'Short code',
+    field: 'shortCode',
+  },
+  {
+    title: 'Description',
+    field: 'description',
+  },
+  {
+    title: 'Instruments',
+    field: 'instruments.length',
+    emptyValue: '-',
+  },
+];
+
+const TechniqueTable = () => {
+  const {
+    loadingTechniques,
+    techniques,
+    setTechniquesWithLoading: setTechniques,
+  } = useTechniquesData();
+
+  const { api } = useDataApiWithFeedback();
+  const isUserOfficer = useCheckAccess([UserRole.USER_OFFICER]);
+  const [urlQueryParams, setUrlQueryParams] =
+    useQueryParams<UrlQueryParamsType>(DefaultQueryParams);
+
+  const [openTechniqueAssignment, setOpenTechniqueAssignment] = useState(false);
+  const [selectedTechniques, setSelectedTechniques] = useState<
+    TechniqueFragment[]
+  >([]);
+
+  const onTechniqueDelete = async (techniqueDeletedId: number | string) => {
+    try {
+      await api({
+        toastSuccessMessage: 'Technique deleted successfully!',
+      }).deleteTechnique({
+        id: techniqueDeletedId as number,
+      });
+
+      return true;
+    } catch (error) {
+      return false;
+    }
+  };
+
+  const AssignmentIndIcon = (): JSX.Element => <AssignmentInd />;
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  function setAssigningTechniqueId(technique: TechniqueFragment): void {
+    const techniqueList: TechniqueFragment[] = [];
+    techniqueList.push(technique);
+    setSelectedTechniques(techniqueList);
+    setOpenTechniqueAssignment(true);
+  }
+
+  const assignInstrumentsToTechniques = async (
+    instruments: InstrumentFragment[]
+  ): Promise<void> => {
+    const techniqueId = selectedTechniques[0].id;
```

Comment:
Kindly help me understand this. 

Are we going to attach the instruments only to the first selected technique? That means, the remaining techniques will not be attached to the instruments?